llvm/include/llvm/Transforms/IPO/Attributor.h

```diff
@@ -5817,8 +5817,12 @@ struct AAPointerInfo : public AbstractAttribute {
     /// Copy ranges from \p L that are not in \p R, into \p D.
     static void set_difference(const RangeList &L, const RangeList &R,
                                RangeList &D) {
-      std::set_difference(L.begin(), L.end(), R.begin(), R.end(),
-                          std::back_inserter(D), RangeTy::OffsetLessThan);
+      std::set_difference(
+          L.begin(), L.end(), R.begin(), R.end(), std::back_inserter(D),
+          [](const RangeTy &L, const RangeTy &R) {
+            return (L.Offset < R.Offset) ||
+                   ((L.Offset == R.Offset) && (L.Size != R.Size));
```

Comment:
std::set_difference() expects a "less than" comparison operator ... to that effect, this line needs to return `L.Size < R.Size`. Can we additionally just rename RangeTy::OffsetLessThan to RangeTy::LessThan, and then upgrade it by putting this entire check there?

Reply:
I tried that but it broke more things, such as adding plenty of entries with same offset but different sizes into either `AccessList` or `OffsetBins`.

Reply:
I didn't do it correctly before. This does work. Updated.